File: android-app/core/model/src/main/kotlin/com/prayer/pointfinder/core/model/Stage.kt
package com.prayer.pointfinder.core.model

import kotlinx.serialization.Serializable

@Serializable
data class Stage(
    val id: EntityId,
    val gameId: EntityId? = null,
    val name: String,
    val description: String = "",
    val orderIndex: Int = 0,
    val transitionType: String = "manual",
    val scheduledAt: String? = null,
    val triggerBaseId: EntityId? = null,
    val isActive: Boolean = false,
    val baseIds: List<EntityId>? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
)

@Serializable
data class CreateStageRequest(
    val name: String,
    val description: String? = null,
    val transitionType: String = "manual",
    val scheduledAt: String? = null,
    val triggerBaseId: EntityId? = null,
)

@Serializable
data class UpdateStageRequest(
    val name: String,
    val description: String? = null,
    val transitionType: String = "manual",
    val scheduledAt: String? = null,
    val triggerBaseId: EntityId? = null,
)

@Serializable
data class ReorderStagesRequest(val ids: List<EntityId>)

File: web/src/test/factories/stage.ts
import type { Stage } from '@/types/stage'

let counter = 0

export function createMockStage(overrides?: Partial<Stage>): Stage {
  counter++
  return {
    id: `stage-${counter}`,
    gameId: 'game-1',
    name: `Stage ${counter}`,
    description: null,
    orderIndex: counter - 1,
    transitionType: 'manual',
    scheduledAt: null,
    triggerBaseId: null,
    isActive: counter === 1,
    baseIds: [],
    createdAt: new Date().toISOString(),
    updatedAt: new Date().toISOString(),
    ...overrides,
  }
}

export function resetStageCounter() {
  counter = 0
}
